src/assert.js

```diff
@@ -26,6 +26,7 @@ QUnit.assert = Assert.prototype = {
 			acceptCallCount = 1;
 		}
 
+		test.asyncCallCount += acceptCallCount;
```

Comment:
Since `semaphore` could potentially be modified elsewhere, this helps us keep track of the number of `assert.async` calls. Unsure if that is a legitimate concern or not, as it was difficult for me to follow all the paths where `semaphore` might be modified.